### PR TITLE
feat: Trailing commas for `with ...` lists and `init()` parameters

### LINKED ISSUE #245: Support trailing commas for remaining comma-separated lists
This is a follow-up to #179

`grammar.ohm` lacks support for trailing commas:
* `init()` parameters
* list of inherited traits (`with ...`)

Note, that as `receive()`, `bounced()` and `external()` have at most one parameter, it probably doesn't make sense to add trailing commas to them.

### LINKED ISSUE #247: Do not allow empty lists of inherited traits
So, something like this should be disabled:

```tact
contract Foo with {
}
```

### PR DESCRIPTION
**Pull request by novusnota**
Closes #245
Closes #247

Additionally changed one line in`jest.config.js` (from `maxWorkers: 1` to `maxWorkers: "50%"`) which gave 7-10s speedup to tests on my machine and should also perform well in CI.

- [x] I have updated CHANGELOG.md
- [x] I have added unit tests to demonstrate the contribution is correctly implemented
- [x] I have run all the tests locally and no test failure was reported
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Update the `dump` function to handle addresses: PR [#175](https://github.com/tact-lang/tact/pull/175)
-- Support trailing commas for struct fields and function arguments: PR [#179](https://github.com/tact-lang/tact/pull/179)
+- Support trailing commas in all comma-separated lists (struct instance fields, `initOf` arguments, `init()` parameters, inherited traits via `with`, function arguments and parameters): PR [#179](https://github.com/tact-lang/tact/pull/179) and PR [#246](https://github.com/tact-lang/tact/pull/246)
```
Comment (anton-trunov) [inline CHANGELOG.md]: Let's also support trailing commas in struct declarations

File: src/grammar/__snapshots__/grammar.spec.ts.snap [hunk 2]
```diff
@@ -254,6 +254,34 @@ Line 2, col 31:
 "
 `;
 
+exports[`grammar should fail case-27 1`] = `
+"<unknown>:1:19: Empty inherited traits list should not have a dangling comma.
```
Comment (anton-trunov) [inline src/grammar/__snapshots__/grammar.spec.ts.snap]: We should disallow empty trait lists entirely. We allow it now, but it's a bug, see #247, so let's fix it here.

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Cool stuff! Let's also allow trailing commas in struct declarations and disallow empty trait lists

File: CHANGELOG.md (hunk 1 above)
Comment (novusnota) [reply CHANGELOG.md]: Hmm, there aren't any — fields of structs and messages end with semi-colons `;`

File: src/grammar/__snapshots__/grammar.spec.ts.snap (hunk 2 above)
Comment (novusnota) [reply src/grammar/__snapshots__/grammar.spec.ts.snap]: Agreed

File: CHANGELOG.md (hunk 1 above)
Comment (anton-trunov) [reply CHANGELOG.md]: argh, you are right -- it's an inconsistency in the grammar design

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**